cmake/kokkos_arch.cmake

```diff
@@ -617,8 +621,9 @@ IF(KOKKOS_ENABLE_SYCL AND NOT BUILD_SHARED_LIBS)
       KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED)
 
     IF(KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED)
+      # Only the separable compilation implementation is supported.
       COMPILER_SPECIFIC_FLAGS(
-        DEFAULT -fsycl-device-code-split=off -DDESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
+        DEFAULT -fsycl-device-code-split=off -DDESUL_ATOMICS_ENABLE_SYCL_SEPARABLE_COMPILATION -DDESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
```

Comment:
How can we capture this setting within Kokkos source code?
(`#ifdef DESUL_ATOMICS_ENABLE_SYCL_SEPARABLE_COMPILATION` is not the answer, I will move these compile definitions to `<Kokkos_Atomics_Desul_Config.hpp>` where there belong in a follow up.)

Also is it not something you would want to see when the configuration is printed?

Reply:
> How can we capture this setting within Kokkos source code?

We might want to define a corresponding Kokkos `CMake` option as we have for `Cuda` and `HIP`, see https://github.com/kokkos/kokkos/pull/5993, which enables some more optimizations on the compiler level but also has some downsides as we are using some low-level functions in `Kokkos` itself.
Without this option, I think it's fine just to configure `desul/atomics/Config.hpp` accordingly.

> Also is it not something you would want to see when the configuration is printed?

Not sure. My gut feeling is to document choices that are somewhat directly done by the user.

Reply:
> > How can we capture this setting within Kokkos source code?
> 
> We might want to define a corresponding Kokkos `CMake` option as we have for `Cuda` and `HIP`, see #5993, which enables some more optimizations on the compiler level but also has some downsides as we are using some low-level functions in `Kokkos` itself. Without this option, I think it's fine just to configure `desul/atomics/Config.hpp` accordingly.

Never mind I realize now we can do
```C++
#if defined(KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED) && defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
```
Let's not get into the `Kokkos_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE` option discussion just yet

Reply:
> Let's not get into the Kokkos_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE option discussion just yet

I agree. 🙂

Reply:
> > Also is it not something you would want to see when the configuration is printed?
> 
> Not sure. My gut feeling is to document choices that are somewhat directly done by the user.

You are missing the point of `Kokkos::print_configuration`.  It is not so much for users it is more for developers to know for sure what was built.
And I expect that knowing whether `SYCL_EXT_ONEAPI_DEVICE_GLOBAL` is defined or not is useful information.